xds/src/main/java/com/linecorp/centraldogma/xds/cluster/v1/XdsClusterService.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.xds.cluster.v1;
+
+import static com.linecorp.centraldogma.server.internal.admin.auth.AuthUtil.currentAuthor;
+import static com.linecorp.centraldogma.xds.internal.ControlPlanePlugin.CLUSTERS_DIRECTORY;
+import static com.linecorp.centraldogma.xds.internal.XdsResourceManager.RESOURCE_ID_PATTERN;
+import static com.linecorp.centraldogma.xds.internal.XdsResourceManager.RESOURCE_ID_PATTERN_STRING;
+import static com.linecorp.centraldogma.xds.internal.XdsResourceManager.removePrefix;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.protobuf.Empty;
+
+import com.linecorp.centraldogma.xds.cluster.v1.XdsClusterServiceGrpc.XdsClusterServiceImplBase;
+import com.linecorp.centraldogma.xds.internal.XdsResourceManager;
+
+import io.envoyproxy.envoy.config.cluster.v3.Cluster;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+
+/**
+ * Service for managing clusters.
+ */
+public final class XdsClusterService extends XdsClusterServiceImplBase {
+
+    private static final Pattern CLUSTER_NAME_PATTERN =
+            Pattern.compile("^groups/([^/]+)/clusters/" + RESOURCE_ID_PATTERN_STRING + '$');
+
+    private final XdsResourceManager xdsResourceManager;
+
+    /**
+     * Creates a new instance.
+     */
+    public XdsClusterService(XdsResourceManager xdsResourceManager) {
+        this.xdsResourceManager = xdsResourceManager;
+    }
+
+    @Override
+    public void createCluster(CreateClusterRequest request, StreamObserver<Cluster> responseObserver) {
+        final String parent = request.getParent();
+        final String group = removePrefix("groups/", parent);
+        xdsResourceManager.checkGroup(group);
```

Comment:
Note: Calling `createGroup` and immediately calling `createCluster` might fail if the created group is not propagated across replicas yet

Reply:
That is correct. 👍 It's like creating a repository and immediately pushing a file might fail.

Reply:
Understood, also noted that the revision isn't returned in the response.